<?xml version="1.0" ?>
<robot name="husky">

  <!-- Mobile Base -->
  <link name="base_link">
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0.061875"/>
      <geometry>
        <box size="0.9874 0.5709 0.12375"/>
      </geometry>
    </collision>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0.17562499999999998"/>
      <geometry>
        <box size="0.7899200000000001 0.5709 0.10375"/>
      </geometry>
    </collision>
  </link>

  <!-- Base footprint is on the ground under the robot -->
  <link name="base_footprint"/>
  <joint name="base_footprint_joint" type="fixed">
    <origin rpy="0 0 0" xyz="0 0 -0.13228"/>
    <parent link="base_link"/>
    <child link="base_footprint"/>
  </joint>

  <!-- Inertial link stores the robot's inertial information -->
  <link name="inertial_link">
    <inertial>
      <mass value="46.034"/>
      <origin xyz="-0.00065 -0.085 0.062"/>
      <inertia ixx="0.6022" ixy="-0.02364" ixz="-0.1197" iyy="1.7386" iyz="-0.001544" izz="2.0296"/>
    </inertial>
  </link>

  <joint name="inertial_joint" type="fixed">
    <origin rpy="0 0 0" xyz="0 0 0"/>
    <parent link="base_link"/>
    <child link="inertial_link"/>
  </joint>

  <link name="left_wheel">
    <inertial>
      <mass value="2.637"/>
      <origin xyz="0 0 0"/>
      <inertia ixx="0.02467" ixy="0" ixz="0" iyy="0.04411" iyz="0" izz="0.02467"/>
    </inertial>
    <collision>
      <origin rpy="1.570795 0 0" xyz="0 0 0"/>
      <geometry>
        <cylinder length="0.1143" radius="0.1651"/>
      </geometry>
    </collision>
  </link>

  <!-- Pinocchio takes 'continuous' type as nv=2 -->
  <joint name="left_wheel" type="revolute"> 
    <parent link="base_link"/>
    <child link="left_wheel"/>
    <origin rpy="0 0 0" xyz="0 0.2775 0.03282"/>
    <axis rpy="0 0 0" xyz="0 1 0"/>
    <limit effort="100.0" lower="-3.141592" upper="3.141592" velocity="1000"/>
  </joint>

  <link name="right_wheel">
    <inertial>
      <mass value="2.637"/>
      <origin xyz="0 0 0"/>
      <inertia ixx="0.02467" ixy="0" ixz="0" iyy="0.04411" iyz="0" izz="0.02467"/>
    </inertial>
    <collision>
      <origin rpy="1.570795 0 0" xyz="0 0 0"/>
      <geometry>
        <cylinder length="0.1143" radius="0.1651"/>
      </geometry>
    </collision>
  </link>

  <joint name="right_wheel" type="revolute">
    <parent link="base_link"/>
    <child link="right_wheel"/>
    <origin rpy="0 0 0" xyz="0 -0.2775 0.03282"/>
    <axis rpy="0 0 0" xyz="0 1 0"/>
    <limit effort="100.0" lower="-3.141592" upper="3.141592" velocity="1000"/>
  </joint>

  <link name="top_structure_link">
    <collision>
      <origin rpy="0 0 0" xyz="-0.2 0 0.85"/>
      <geometry>
        <box size="0.38992 0.56 1.365"/>
      </geometry>
      <inertial>
        <origin rpy="0 0 0" xyz="-0.2 0 0.45"/>
        <mass value="40"/>
        <inertia ixx="1.95397859e-01" ixy="7.23330468e-08" ixz="2.01114568e-02" iyy="1.92679403e-01" iyz="-7.81325404e-05" izz="1.01958752e-01"/>
      </inertial>
    </collision>
  </link>
  <joint name="top_structure_joint" type="fixed">
    <parent link="base_link"/>
    <child link="top_structure_link"/>
    <origin rpy="0 0 0" xyz="0 0 0"/>
  </joint>
</robot>
